Services/TMS/Booking/classes/class.ilTMSSelfBookingGUI.php

```diff
@@ -183,6 +183,16 @@ protected function getConfirmButtonLabel() {
 		return $this->g_lng->txt("booking_confirm");
 	}
 
+	/**
+	 * @inheritdocs
+	 */
+	protected function getApplicableSteps() {
+		$steps = $this->getComponentsOfType(Booking\SelfBookingStep::class);
```

Comment:
Bitte statt der Überschreibung einfach den `Booking\Player` nach der Kompontenklasse parametrisieren. D.h. eine `abstract function getComponentClass` auf dem `Booking\Player` einführen und diese in `Booking\Player::getApplicableSteps` verwenden.